### DISCUSSION TITLE
Recovery code question

### DISCUSSION THREAD

**Discussion by MurphyDoe0815** [dani-garcia/vaultwarden#3050] [Q&A] [answered]
hello everyone

i have activated a 2fa factor. if the phone gets away now also copied the recovery code.

now i wanted to try the recovery code, but i can't get past the link to "select another method".
here i click on the link for recovery code , but it only ever opens the wiki page ?
what am i doing wrong ?

docker/Version 2022.12.0

**Comment by BlackDex** [comment, marked as answer]
If you read that page carefully, then it tells you where to go by typing the correct url.

**Comment by MurphyDoe0815** [comment]
You are right - those who can read are at a clear advantage.

but i'm sorry to say that it's not intuitive. 
the upper link leads to 2FA and the lower one, which looks the same, leads to the wiki page that explains how to enter it manually.
I was then out of the loop 

Thank you for the quick help !

**Reply by BlackDex** [reply]
I suggest to report this at Bitwarden, but i think there already is something regarding this. I have no idea why they have this flow though.

**Reply by tessus** [reply]
I agree, the flow is not very smart. Unless they want to track how many people have to use a recovery code. ;-)

**Reply by MurphyDoe0815** [reply]
what would be good of course, you could download the qr code of the 2fa and save it. then you would not have to have an additional recovery code. i find the screenshot taking really cumbersome

**Reply by BlackDex** [reply]
That wouldn't be really secure though.
Better to save the secret string somewhere save (other than in Vaultwarden).
Or, what i do, i have multiple MFA apps on my phone, which make secure backups.

**Reply by MurphyDoe0815** [reply]
I see it differently:
1. to get to the container, I have to make vaultwarden with port 443 available to the network.
- Here it would be enough for me to access it locally and via vpn from outside. 
If it is on the network - I am vulnerable - but there is no other way !!!!
2. what is the difference between saving the recovery code or the QR code?
If my mobile phone is gone or broken, only the recovery code would be important. And where do I save it, where I don't also save the screenshot ? 

I have now tried out a trial account and played through all the variants. I still come to the conclusion that I save the recovery code and / or the QR code locally - without internet access.

Nextcloud with the "Aegis" app and only (!) the QR code would be my favourite.
Nextcloud does not necessarily require internet access ; -)

**Reply by BlackDex** [reply]
Well, i suggest to create a suggest to Bitwarden in this case. They build and maintain the clients, including the web-vault.
It's the same on there vault.bitwarden.com so it isn't a Vaultwarden only item.

**Reply by MurphyDoe0815** [reply]
i don't think this will be successful. Maybe someone will read it and follow it up. 

Thank you for your constructive answers ;-)

**Reply by BlackDex** [reply]
I actually doubt someone from Bitwarden will read it here and follow up on it.
Anyway, there clients are also open source, and people can create PR's there too.
So if, then people can contribute to Bitwarden that way also.
